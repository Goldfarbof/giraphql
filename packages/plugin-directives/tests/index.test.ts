import {
  printSchema,
  lexicographicSortSchema,
  GraphQLObjectType,
  GraphQLEnumType,
  GraphQLInputObjectType,
} from 'graphql';
import exampleSchema from './example/schema';

describe('extends example schema', () => {
  test('generates expected schema', () => {
    expect(printSchema(lexicographicSortSchema(exampleSchema))).toMatchSnapshot();
  });

  test('has expected directives in extensions', () => {
    const types = exampleSchema.getTypeMap();

    expect(types.Obj.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 'o' }]);
    expect(types.Query.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 'o' }]);

    const testField = (types.Query as GraphQLObjectType).getFields().test;

    expect(testField.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 'f' }]);
    expect(testField.args[0].extensions?.directives).toStrictEqual([
      { args: { foo: 123 }, name: 'a' },
    ]);

    expect(types.IF.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 'i' }]);
    expect(types.UN.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 'u' }]);
    expect(types.EN.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 'e' }]);
    expect((types.EN as GraphQLEnumType).getValue('ONE')?.extensions?.directives).toStrictEqual([
      { args: { foo: 123 }, name: 'ev' },
    ]);
    expect(types.Date.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 's' }]);
    expect(types.In.extensions?.directives).toStrictEqual([{ args: { foo: 123 }, name: 'io' }]);
    expect(
      (types.In as GraphQLInputObjectType).getFields().test.extensions?.directives,
    ).toStrictEqual([{ args: { foo: 123 }, name: 'if' }]);
  });
});
