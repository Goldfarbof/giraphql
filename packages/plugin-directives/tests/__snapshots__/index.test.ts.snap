// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extends example schema generates expected schema 1`] = `
"scalar Date

enum EN {
  ONE
}

interface IF

input In {
  test: String
}

type Obj

type Query {
  test(arg1: String): String!
}

union UN
"
`;
